### PR TITLE
CD-412

### PR DESCRIPTION
**Pull request by PivaFiras**
- Add ESXi General Information script
- Add script description JSON

### CODE REVIEW

File: examples/http/esxi_general_information.js [hunk 1]
```diff
@@ -0,0 +1,267 @@
...
+ */
+function generateVariables(soapResponse) {
+  const $ = D.htmlParse(soapResponse);
+
+  // Helper function to extract the value of a property from the parsed response based on the given path.
+  function getPropSetValue(path) {
+    return $("propSet").has("name:contains('" + path + "')").find("val").text() || "N/A";
+  }
+
+  const variables = [
```
Comment (mohamedkallel) [inline examples/http/esxi_general_information.js:213]: put this as global variable and use it to generate the retreive properties envelope

File: examples/http/esxi_general_information.js [hunk 2]
```diff
@@ -0,0 +1,267 @@
...
+const url = '/sdk'
+
+// Variable to store the headers for the SOAP request. This will include the vmware_soap_session.
+let headers = "";
+
+/**
+ * Sets the request headers with the provided cookie information.
+ * @param {Object} headerCookie - An object containing the 'Cookie' header with the appropriate value.
+ */
+function setHeaders(headerCookie) {
```
Comment (mohamedkallel) [inline examples/http/esxi_general_information.js:34]: why do not use jar: true parameter instead of setting the headers manually

**Review by mohamedkallel — CHANGES_REQUESTED** [review]

File: examples/http/esxi_general_information.js [hunk 3]
```diff
@@ -0,0 +1,257 @@
...
+ * @returns {Promise} A promise that resolves with the properties of the host as extracted from the SOAP response.
+ */
+function retrieveProprieties(hostRef) {
+  const payload = createSoapPayload(
+      '<vim25:RetrieveProperties>' +
+      '   <vim25:_this type="PropertyCollector">ha-property-collector</vim25:_this>' +
+      '   <vim25:specSet>' +
+      '      <vim25:propSet>' +
+      '         <vim25:type>HostSystem</vim25:type>' +
+      '         <vim25:pathSet>runtime.connectionState</vim25:pathSet>' +
```
Comment (mohamedkallel) [inline examples/http/esxi_general_information.js:178]: please use the defined global variable "variables" that you created

**Review by mohamedkallel — CHANGES_REQUESTED** [review]

File: examples/http/esxi_general_information.js [hunk 4]
```diff
@@ -0,0 +1,260 @@
...
+function getSessionKey(soapResponse) {
+  const $ = D.htmlParse(soapResponse)
+  return $('returnval').find('key').first().text()
+}
+
+/**
+ * Constructs and sends a SOAP login request to the ESXi server.
+ * @returns {Promise} A promise that resolves with the response of the login request.
+ */
+function login () {
```
Comment (GeorgiTsatsev) [inline examples/http/esxi_general_information.js:101]: what if the login is unsuccessful ? There is no proper handling of that (it must call the D.failure callback with the relevant error - D.errorType.AUTHENTICATION_ERROR )

**Review by GeorgiTsatsev — APPROVED** [review]

**Pull request merged**